### DISCUSSION TITLE
Working with mixed html and markdown

### DISCUSSION THREAD

**Discussion by henryscott-wk** [mkdocs/mkdocs#2495] [Q&A]
Does mkdocs support using mixed html and markdown? I.e. can I link to certain static pages within my markdown, and if I do will it be indexed with the search_index.json?

**Comment by oprypin** [comment]
Short story, no. But, Markdown is a superset of HTML so you can just use HTML in *.md files. But that wouldn't become whole-page HTML, but rather just take up the body part. However, I think you could then replace templates for some pages to become your custom HTML, including empty HTML, then perhaps those *.md files become whole-page replacements, if you need that.

**Reply by henryscott-wk** [reply]
Makes sense to me. I'm avoiding doing anything too hackey, but I'm interested in seeing what I can do to integrate a Javadoc generated site into a tool that uses mkdocs for its rendering. It doesn't sound especially well supported, so no proof of concept isn't catastrophic

**Reply by oprypin** [reply]
@henryscott-wk For full proper support, if you're willing to put in the work, you could integrate a pluggable Java doc renderer as a [mkdocstrings handler](https://mkdocstrings.github.io/#features).
